python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -1313,6 +1312,15 @@ def _test_write_to_dataset_with_partitions(base_path,
     # Partitioned columns become 'categorical' dtypes
     for col in partition_by:
         output_df[col] = output_df[col].astype('category')
+
+    if schema:
+        expected_date_type = schema.field_by_name('date').type.to_pandas_dtype()
+    else:
+        # Arrow to Pandas v2 will convert date32 to [ms]. Pandas v1 will always
+        # silently coerce to [ns] due to non-[ns] support.
+        expected_date_type = 'datetime64[ms]'
+    output_df["date"] = output_df["date"].astype(expected_date_type)
```

Comment:
There is no issue with reading datsets, I just had to update this test to check if the resulting pandas df matched the overridden schema types if a schema is applied. There is probably a better way to check then what I've done here..